src/main/java/org/mockito/internal/matchers/CustomCapturingMatcher.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.matchers;
+
+import java.io.Serializable;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.mockito.ArgumentMatcher;
+import org.mockito.exceptions.Reporter;
+
+public class CustomCapturingMatcher<T> implements ArgumentMatcher<T>, ArgumentHolder<T>, CapturesArguments, VarargMatcher, Serializable {
```

Comment:
I think i is not neccessary to have a `CustomCapturingMatcher`,  this implementation could be moved to `CapturingMatcher` and use an `any() `Matcher if used in a `ArgumentCaptor.capture()` call.